Index: strongswan/src/libstrongswan/plugins/openssl/openssl_kdf.c
===================================================================
--- strongswan.orig/src/libstrongswan/plugins/openssl/openssl_kdf.c
+++ strongswan/src/libstrongswan/plugins/openssl/openssl_kdf.c
@@ -199,6 +199,9 @@ kdf_t *openssl_kdf_create(key_derivation
 		/* use a lengthy key to test the implementation below to make sure the
 		 * algorithms are usable, see openssl_hmac.c for details */
 		.key = chunk_clone(chunk_from_str("00000000000000000000000000000000")),
+		.salt = chunk_clone(chunk_from_chars(
+			    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+			    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
 	);
 
 	/* also generate a salt (as if none was provided, i.e. zeroes of hash length)
