From: Yves-Alexis Perez <corsac@debian.org>
Date: Sat, 30 Apr 2022 15:28:32 +0200
Subject: fix format string issue in enum_flags_to_string()

---
 src/libstrongswan/utils/enum.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libstrongswan/utils/enum.c b/src/libstrongswan/utils/enum.c
index 79da450..1e77489 100644
--- a/src/libstrongswan/utils/enum.c
+++ b/src/libstrongswan/utils/enum.c
@@ -97,7 +97,7 @@ char *enum_flags_to_string(enum_name_t *e, u_int val, char *buf, size_t len)
 		return buf;
 	}
 
-	if (snprintf(buf, len, e->names[0]) >= len)
+	if (snprintf(buf, len, "%s", e->names[0]) >= len)
 	{
 		return NULL;
 	}
