diff --git a/src/pki/commands/issue.c b/src/pki/commands/issue.c
index 20163ed..627fb7c 100644
--- a/src/pki/commands/issue.c
+++ b/src/pki/commands/issue.c
@@ -72,7 +72,7 @@ static int issue()
 	chunk_t serial = chunk_empty;
 	chunk_t encoding = chunk_empty;
 	time_t lifetime = 1095;
-	time_t not_before, not_after;
+	time_t not_before = time(NULL), not_after;
 	x509_flag_t flags = 0;
 	x509_t *x509;
 	x509_cdp_t *cdp = NULL;
@@ -138,6 +138,14 @@ static int issue()
 					goto usage;
 				}
 				continue;
+			case 'S':
+				not_before = atoi(arg);
+				if (!not_before)
+				{
+					error = "invalid --not-before value";
+					goto usage;
+				}
+				continue;
 			case 's':
 				hex = arg;
 				continue;
@@ -431,7 +439,6 @@ static int issue()
 										chunk_from_chars(ASN1_SEQUENCE, 0));
 	}
 
-	not_before = time(NULL);
 	not_after = not_before + lifetime * 24 * 60 * 60;
 
 	cert = lib->creds->create(lib->creds, CRED_CERTIFICATE, CERT_X509,
@@ -518,7 +525,7 @@ static void __attribute__ ((constructor))reg()
 		 "[--cert-policy oid [--cps-uri uri] [--user-notice text] ]+",
 		 "[--policy-map issuer-oid:subject-oid]",
 		 "[--policy-explicit len] [--policy-inhibit len] [--policy-any len]",
-		 "[--digest md5|sha1|sha224|sha256|sha384|sha512] [--outform der|pem]"},
+		 "[--digest md5|sha1|sha224|sha256|sha384|sha512] [--outform der|pem] [--not-before seconds-since-epoch]"},
 		{
 			{"help",			'h', 0, "show usage information"},
 			{"in",				'i', 1, "public key/request file to issue, default: stdin"},
@@ -547,6 +554,7 @@ static void __attribute__ ((constructor))reg()
 			{"ocsp",			'o', 1, "OCSP AuthorityInfoAccess URI to include"},
 			{"digest",			'g', 1, "digest for signature creation, default: sha1"},
 			{"outform",			'f', 1, "encoding of generated cert, default: der"},
+			{"not-before",			'S', 1, "start time of certificate (in seconds since the epoch), default: now"},
 		}
 	});
 }
diff --git a/src/pki/commands/self.c b/src/pki/commands/self.c
index c4508a6..46fa6cf 100644
--- a/src/pki/commands/self.c
+++ b/src/pki/commands/self.c
@@ -60,7 +60,7 @@ static int self()
 	chunk_t serial = chunk_empty;
 	chunk_t encoding = chunk_empty;
 	time_t lifetime = 1095;
-	time_t not_before, not_after;
+	time_t not_before = time(NULL), not_after;
 	x509_flag_t flags = 0;
 	x509_cert_policy_t *policy = NULL;
 	char *arg;
@@ -121,6 +121,14 @@ static int self()
 					goto usage;
 				}
 				continue;
+			case 'S':
+				not_before = atoi(arg);
+				if (!not_before)
+				{
+					error = "invalid --not-before value";
+					goto usage;
+				}
+				continue;
 			case 's':
 				hex = arg;
 				continue;
@@ -302,7 +310,6 @@ static int self()
 		}
 		rng->destroy(rng);
 	}
-	not_before = time(NULL);
 	not_after = not_before + lifetime * 24 * 60 * 60;
 	cert = lib->creds->create(lib->creds, CRED_CERTIFICATE, CERT_X509,
 						BUILD_SIGNING_KEY, private, BUILD_PUBLIC_KEY, public,
@@ -382,7 +389,7 @@ static void __attribute__ ((constructor))reg()
 		 "[--cert-policy oid [--cps-uri uri] [--user-notice text] ]+",
 		 "[--policy-map issuer-oid:subject-oid]",
 		 "[--policy-explicit len] [--policy-inhibit len] [--policy-any len]",
-		 "[--digest md5|sha1|sha224|sha256|sha384|sha512] [--outform der|pem]"},
+		 "[--digest md5|sha1|sha224|sha256|sha384|sha512] [--outform der|pem] [--not-before seconds-since-epoch]"},
 		{
 			{"help",			'h', 0, "show usage information"},
 			{"in",				'i', 1, "private key input file, default: stdin"},
@@ -407,6 +414,7 @@ static void __attribute__ ((constructor))reg()
 			{"ocsp",			'o', 1, "OCSP AuthorityInfoAccess URI to include"},
 			{"digest",			'g', 1, "digest for signature creation, default: sha1"},
 			{"outform",			'f', 1, "encoding of generated cert, default: der"},
+			{"not-before",			'S', 1, "start time of certificate (in seconds since the epoch), default: now"},
 		}
 	});
 }
