diff --git a/src/ipsec/_ipsec.in b/src/ipsec/_ipsec.in
index 3c1f998..00f6a4b 100644
--- a/src/ipsec/_ipsec.in
+++ b/src/ipsec/_ipsec.in
@@ -34,6 +34,7 @@ IPSEC_CONFDIR="@IPSEC_CONFDIR@"
 IPSEC_PIDDIR="@IPSEC_PIDDIR@"
 IPSEC_SCRIPT="@IPSEC_SCRIPT@"
 
+IPSEC_STARTER_PID_OLD="${IPSEC_PIDDIR}/starter.pid"
 IPSEC_STARTER_PID="${IPSEC_PIDDIR}/starter.${DAEMON_NAME}.pid"
 IPSEC_CHARON_PID="${IPSEC_PIDDIR}/${DAEMON_NAME}.pid"
 
@@ -261,10 +262,16 @@ status|statusall)
 	;;
 stop)
 	# stopping a not-running service is considered as success
-	if [ -e $IPSEC_STARTER_PID ]
+	if [ -e $IPSEC_STARTER_PID ] || [ -e $IPSEC_STARTER_PID_OLD ]
 	then
+		pidfile=$IPSEC_STARTER_PID
+		if [ -e $IPSEC_STARTER_PID_OLD ]
+		then
+			pidfile=$IPSEC_STARTER_PID_OLD
+		fi
+
 		echo "Stopping strongSwan IPsec..." >&2
-		spid=`cat $IPSEC_STARTER_PID`
+		spid=`cat $pidfile`
 		if [ -n "$spid" ]
 		then
 			kill $spid 2>/dev/null
@@ -277,7 +284,7 @@ stop)
 			if [ $loop -eq 0 ]
 			then
 				kill -KILL $spid 2>/dev/null
-				rm -f $IPSEC_STARTER_PID
+				rm -f $pidfile
 			fi
 		fi
 	else
