diff --git a/init/systemd/strongswan.service.in b/init/systemd/strongswan.service.in
index 474284a1..86d55304 100644
--- a/init/systemd/strongswan.service.in
+++ b/init/systemd/strongswan.service.in
@@ -4,7 +4,7 @@ After=syslog.target network-online.target
 [Service]
 ExecStart=@SBINDIR@/@IPSEC_SCRIPT@ start --nofork
 ExecReload=@SBINDIR@/@IPSEC_SCRIPT@ reload
-StandardOutput=syslog
+StandardOutput=journal
 Restart=on-abnormal
 
 [Install]
